src/v/config/configuration.cc

```diff
@@ -3785,17 +3785,17 @@ configuration::configuration()
         .visibility = visibility::user,
       },
       std::nullopt)
-  , iceberg_rest_catalog_user_id(
+  , iceberg_rest_catalog_client_id(
```

Comment:
@mmaslankaprv  @mattschumpert was proposing some changes to iceberg_catalog_type in the gdoc.. you may want to plug that change in here (I don't think there is a definite proposal, Matt can you chime in please ). I think we are considering filesystem/bucket/cloud-storage